### ISSUE TITLE
json tooltip data

### ISSUE THREAD

**Issue by remco-pc** [ajaxorg/ace#5391]
### Describe the feature

is there a feature that i can extend / replace the tooltip behaviour to my own custom templates and is this possible to have it stored in json data ?

### Use Case

i would like to add some own functions to the tooltip / filler and ideally it can be stored in json files, so i can import /export them easilly with new features.

### Proposed Solution

have all data placed in json files, it's cleaner code and separation of concerns

### Acknowledgements

- [X] I may be able to implement this feature request
- [X] This feature might incur a breaking change

### ACE version used

1.4.14

**Comment by nightwing** [comment]
Are you talking about autocompletion, or something else? For autocompletion json alone would not be helpful as it requires a parser as well.

**Comment by remco-pc** [comment]
yes i like to feed to autocomplete with json data if possible / edit files, complete files

**Comment by remco-pc** [comment]
can't you separate data from code easily ?

**Comment by nightwing** [comment]
Sorry, i had missed your reply.
The problem here is that in addition to data you need some kind of parser that will figure out which data and where should go. This parser depends on language for which you want to provide autocompletions and in general case is not provided for ace.
https://github.com/mkslanc/ace-linters/ provides autocompletions for many languages, for json and yaml it accepts schema, so maybe it already provides the feature you need.
If it does not, could you please tell me more details of your usecase, and show some example code?

**Comment by remco-pc** [comment]
@nightwing thanks for commenting this out, ace editor is a great editor, but don't have the time (and given priority) to dive deeper into this editor, i am using it for logs and searches on the hd (windows search does not find it on my computer) and it could open 500 MB log files on my browser. any change to increase this in the future too or should log files per day not be bigger then 500 MB or make just log files of 500 MB max and start with a new file ?
